src/main/java/io/nats/service/Service.java

```diff
@@ -109,19 +106,17 @@ public void addServiceEndpoint(ServiceEndpoint se) {
             ctx = new EndpointContext(conn, null, false, se);
         }
         serviceContexts.put(se.getName(), ctx);
+
+        // if the service is already started, start the newly added context
         startStopLock.lock();
         try {
             if (runningIndicator != null) {
                 ctx.start();
             }
-        } finally {
-            startStopLock.unlock();
         }
-
-        if (null != infoResponse) {
-            infoResponse.addServiceEndpoint(se);
```

Comment:
infoResponse will never be null since it's initialized first in the constructor.